src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseRepositoryTests.cs

```diff
@@ -393,6 +393,43 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbC
             }
         }
 
+        [Fact]
+        public async Task GetAllReleaseVersionIds()
+        {
+            var originalRelease = new Release();
+
+            var amendedRelease1 = new Release
+            {
+                PreviousVersion = originalRelease,
+            };
+
+            var amendedRelease2 = new Release
+            {
+                PreviousVersion = amendedRelease1,
+            };
+
+            var ignoredRelease = new Release();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(
+                    originalRelease, amendedRelease1, amendedRelease2, ignoredRelease);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var releaseRepository = BuildReleaseRepository(contentDbContext: contentDbContext);
+                var result = await releaseRepository.GetAllReleaseVersionIds(amendedRelease2);
+                Assert.Equal(3, result.Count);
+
+                Assert.Contains(originalRelease.Id, result);
+                Assert.Contains(amendedRelease1.Id, result);
+                Assert.Contains(amendedRelease2.Id, result);
+            }
+        }
```

Comment:
I think you could add an additional test here for a release with no previous version id to check it can return a single version successfully.